pkg/sync/status.go

```diff
@@ -2,110 +2,111 @@ package sync
 
 import (
 	"fmt"
+	"strings"
 
-	"github.com/argoproj/gitops-engine/pkg/health"
-	"github.com/argoproj/gitops-engine/pkg/sync/common"
-	"github.com/argoproj/gitops-engine/pkg/utils/kube"
 	console "github.com/pluralsh/console-client-go"
-	"github.com/samber/lo"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/cli-utils/pkg/apply/event"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	"sigs.k8s.io/cli-utils/pkg/object"
+	"sigs.k8s.io/cli-utils/pkg/print/stats"
 )
 
-func errorAttributes(source string, err error) *console.ServiceErrorAttributes {
-	if err == nil {
-		return nil
+func (engine *Engine) UpdateStatus(id string, ch <-chan event.Event, printStatus bool) error {
+	var statsCollector stats.Stats
+	var err error
+	components := []*console.ComponentAttributes{}
+	statusCollector := &StatusCollector{
+		latestStatus: make(map[object.ObjMetadata]event.StatusEvent),
 	}
 
-	return &console.ServiceErrorAttributes{
-		Source:  source,
-		Message: err.Error(),
+	for e := range ch {
+		statsCollector.Handle(e)
+		switch e.Type {
+		case event.ApplyType:
+			gk := e.ApplyEvent.Identifier.GroupKind
+			name := e.ApplyEvent.Identifier.Name
+			if e.ApplyEvent.Error != nil {
+				msg := fmt.Sprintf("%s apply %s: %s\n", resourceIDToString(gk, name),
+					strings.ToLower(e.ApplyEvent.Status.String()), e.ApplyEvent.Error.Error())
+				if e.ApplyEvent.Status == event.ApplyFailed {
+					err = fmt.Errorf(msg)
```

Comment:
these should be collected and passed as service errors in the updateComponents calls

Reply:
this error is passed to service errors. It is passed here:
```
	if err := engine.updateStatus(id, components, errorAttributes("sync", err)); err != nil {
		log.Error(err, "Failed to update service status, ignoring for now")
	}
```